lib/serial_terminal.pm

```diff
@@ -232,4 +236,37 @@ sub upload_file {
       or die("Failed to finally copy file from '$tmpfilename' to '$dst'");
 }
 
+
+sub reboot {
+    my (%args) = @_;
+    $args{check} //= 1;
+    $args{console} //= testapi::current_console;
+    $args{reboot_cmd} //= 'systemctl reboot';
+    $args{timeout} //= 300;
+    $args{minimum_boot_time} = 3;
+
+    bmwqemu::log_call(%args);
+
+    my $uptime_start;
+    my $time_start = time();
+
+    die('Only root-virtio-terminal is supported') unless $args{console} eq 'root-virtio-terminal';
+    $uptime_start = script_output(q(cat /proc/uptime | awk '{print $1}')) if ($args{check});
+
+    record_info('REBOOT', "cmd: " . $args{reboot_cmd} . ($uptime_start ? "\nuptime: $uptime_start sec" : ''));
+    script_run($args{reboot_cmd}, timeout => 0);
+    reset_consoles;
+    wait_serial('[lL]ogin:', timeout => $args{timeout});
+    select_console($args{console});
+
+    my $duration = time() - $time_start;
```

Comment:
please move this line before selecting console , so we are sure that we not hanging somewhere in openQA side